modules/core/src/lib/layer-extension.js

```diff
@@ -32,13 +32,39 @@ export class LayerExtension {
     return this.constructor === extension.constructor && deepEqual(this.opts, extension.opts);
   }
 
-  getShaders(opts) {
+  getShaders(extension) {
     return null;
   }
 
-  initializeState(context, opts) {}
+  getSubLayerProps(extension) {
+    const {defaultProps = {}} = extension.constructor;
+    const newProps = {
+      updateTriggers: {}
+    };
 
-  updateState(params, opts) {}
+    /* eslint-disable max-depth */
+    for (const key in defaultProps) {
+      if (key in this.props) {
+        const propDef = defaultProps[key];
+        const propValue = this.props[key];
+        newProps[key] = propValue;
+        if (propDef && propDef.type === 'accessor') {
+          newProps.updateTriggers[key] = this.props.updateTriggers[key];
+          // Unwrap sublayer objects before passing to accessor
```

Comment:
Just my 2 cents:

This adds a "hidden" `unwrapObject` method to some layers which feels a little iffy (Layer is already complex and contains a ton of methods some of which are ripe for cleanup). 

The comment here didn't help me understand what "unwrap" means or why it is needed. Also could be valuable to point out if this just a hack for extensions or something more generically valuable for CompositeLayers? Should this be documented at some point? At least add some form of TODO?